### PR TITLE
Term detail screen

### LINKED ISSUE #172: Search App: Term detail screen
Similar to other content types, WordPress terms need a detail screen for being able to view and edit them.

### PR DESCRIPTION
**Pull request by fastlinemedia**
Resolves #172

### CODE REVIEW

File: src/system/ui/forms/index.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ import {
 	SelectItem,
 	SuggestItem,
 	TaxonomyTermsItem,
+	ParentTermItems,
```
Comment (fastlinemedia) [inline src/system/ui/forms/index.js]: Is `ParentTermItems` item still being used? I'm not seeing a component for this and the term form is just using a standard select for the parent option.

File: src/system/ui/forms/index.js [hunk 2]
```diff
@@ -96,6 +97,9 @@ Form.TextareaItem.displayName = 'Form.TextareaItem'
 Form.UrlItem = UrlItem
 Form.UrlItem.displayName = 'Form.UrlItem'
 
+Form.ParentTermItems = ParentTermItems
+Form.ParentTermItems.displayName = 'Form.ParentTermItems'
```
Comment (fastlinemedia) [inline src/system/ui/forms/index.js]: Same as above regarding `ParentTermItems`.

File: src/system/ui/forms/items/taxonomy-terms/index.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ export const TaxonomyTermsItem = ( {
 	taxonomy,
 	value,
 	onChange,
+	isparentSelector,
```
Comment (fastlinemedia) [inline src/system/ui/forms/items/taxonomy-terms/index.js]: Variable should be camelCase.

File: src/system/ui/forms/items/taxonomy-terms/index.js [hunk 4]
```diff
@@ -186,3 +187,63 @@ export const TaxonomyTermsItem = ( {
 		/>
 	)
 }
+
+
+export const ParentTermItems = ( taxonomy,id ) => {
```
Comment (fastlinemedia) [inline src/system/ui/forms/items/taxonomy-terms/index.js]: I'm seeing this now. Form items should be setup as components that can be used in the form config like so...

https://share.getcloudapp.com/qGuDBEKD

Let's break this out into its own file in `form/items` as well.

File: src/system/ui/forms/index.js (hunk 1 above)
Comment (shahidajmeri786) [reply src/system/ui/forms/index.js]: Yes it is used here:
image

File: src/system/ui/forms/items/taxonomy-terms/index.js (hunk 3 above)
Comment (shahidajmeri786) [reply src/system/ui/forms/items/taxonomy-terms/index.js]: Variable not needed anymore.

File: src/system/ui/pages/term/index.js [hunk 5]
```diff
@@ -1,40 +1,168 @@
...
+
+
+		sections: {
+			details: {
+				label: __( 'Details' ),
+				fields: {
+					title: {
+						label: __( 'Name' ),
+						labelPlacement: 'above',
+						value: title
```
Comment (fastlinemedia) [inline src/system/ui/pages/term/index.js]: Form config doesn't accept a value (you may have seen this in old code). To set the initial value, use the `defaults` object below.

File: src/system/ui/pages/term/index.js [hunk 6]
```diff
@@ -1,40 +1,168 @@
...
+					},
+					slug: {
+						label: __( 'Slug' ),
+						labelPlacement: 'above',
+						value: slug,
+					},
+					parent: {
+						label: __( 'Parent' ),
+						labelPlacement: 'above',
+						component: 'parent-terms',
```
Comment (fastlinemedia) [inline src/system/ui/pages/term/index.js]: Looking good! However, it doesn't look like that component was pushed as I'm getting an error when I try to build this.

**Comment by fastlinemedia** [comment]
From closed PR...

**Parent Select**
Added the label placement be beside the select.
Added Only hierarchical terms should show the parent select.
Added the options show the hierarchy?

**Post Count**
Added the label placement be beside the count.

**Actions**
Added action buttons for:
View Archive
Edit in Admin
Delete

**Comment by fastlinemedia** [comment]
Nice work @shahidajmeri786. Everything looks great. The only issue I'm seeing is a unique key error coming from the parent term items select...

https://share.getcloudapp.com/yAuLqn0y

Once that's fixed, we can merge this.

**Comment by shahidajmeri786** [comment]
I checked but I did not get any key error. 
image

**Comment by fastlinemedia** [comment]
> I checked but I did not get any key error.

I wonder if that error doesn't show if you don't have react dev tools installed? Anyhow, it comes from when you map an array to output elements (e.g. with select options). That is fixed by adding a key to each option like so...

https://share.getcloudapp.com/WnuEO5AJ

I pushed that fix and will get this merged. Thanks!

**Pull request merged**